flink-runtime/src/main/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManager.java

```diff
@@ -52,18 +61,45 @@
 public class ResourceManager extends RpcEndpoint<ResourceManagerGateway> {
 	private final ExecutionContext executionContext;
 	private final Map<JobMasterGateway, InstanceID> jobMasterGateways;
+	private final Map<ResourceID, TaskExecutorGateway> taskExecutorGateways;
+	private final Map<ResourceID, ResourceManagerToTaskExecutorHeartbeatScheduler> heartbeatSchedulers;
```

Comment:
Maybe it makes sense to create a `HeartbeatManager` which manages the different resources to heartbeat.